pipecat-client-android-small-webrtc-transport/src/main/java/ai/pipecat/client/small_webrtc_transport/SmallWebRTCTransport.kt

```diff
@@ -0,0 +1,274 @@
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+import kotlinx.serialization.json.decodeFromJsonElement
+
+
+private val BOT_PARTICIPANT = Participant(
+    id = ParticipantId("bot"),
+    name = null,
+    local = false
+)
+
+private val LOCAL_PARTICIPANT = Participant(
+    id = ParticipantId("local"),
+    name = null,
+    local = true
+)
+
+class SmallWebRTCTransport internal constructor(
+    private val transportContext: TransportContext,
+    androidContext: Context,
+    private val serverUrl: String,
+) : Transport() {
+
+    companion object {
+        private const val TAG = "SmallWebRTCTransport"
+    }
+
+    object AudioDevices {
+        val Speakerphone = MediaDeviceInfo(
+            id = MediaDeviceId("speakerphone"),
+            name = "Speakerphone"
+        )
+
+        val Earpiece = MediaDeviceInfo(
+            id = MediaDeviceId("earpiece"),
+            name = "Earpiece"
+        )
+    }
+
+    object Cameras {
+        val Front = MediaDeviceInfo(id = MediaDeviceId("cam-front"), name = "Front Camera")
+        val Rear = MediaDeviceInfo(id = MediaDeviceId("cam-rear"), name = "Rear Camera")
+    }
+
+    class Factory(
+        private val androidContext: Context,
+        private val serverUrl: String
+    ) : TransportFactory {
+        override fun createTransport(context: TransportContext): Transport {
+            return SmallWebRTCTransport(context, androidContext, serverUrl)
+        }
+    }
+
+    private var state = TransportState.Disconnected
+
+    private val appContext = androidContext.applicationContext
+    private val thread = transportContext.thread
+
+    private var client: WebRTCClient? = null
+    private var selectedCam = CameraMode.Front
+
+    override fun initDevices(): Future<Unit, RTVIError> = resolvedPromiseOk(thread, Unit)
+
+    @SuppressLint("MissingPermission")
+    override fun connect(authBundle: AuthBundle?): Future<Unit, RTVIError> =
+        thread.runOnThreadReturningFuture {
+
+            Log.i(TAG, "connect(${authBundle})")
+
+            if (client != null) {
+                return@runOnThreadReturningFuture resolvedPromiseErr(
+                    thread,
+                    RTVIError.OtherError("Connection already active")
+                )
+            }
+
+
+            transportContext.callbacks.onInputsUpdated(
+                camera = false,
+                mic = transportContext.options.enableMic
+            )
+
+            setState(TransportState.Connecting)
+
+            try {
+                client = WebRTCClient(
+                    onIncomingEvent = { msgJson ->
+
+                        val msgWithType =
+                            JSON_INSTANCE.decodeFromJsonElement<MessageWithType>(msgJson)
+
+                        if (msgWithType.type == "signalling") {
+                            val msg = JSON_INSTANCE.decodeFromJsonElement<SignallingEvent>(msgJson)
+
+                            when (msg.message.type) {
```

Comment:
We have also the "renegotiate" message, where the server asks the client to renegotiate.

Reply:
The `renegotiate` one, it is:
- https://github.com/pipecat-ai/pipecat-client-ios-small-webrtc/blob/a6e4516b1fcbed772ca97a9616dddc9329097958/Sources/PipecatClientIOSSmallWebrtc/SmallWebRTCTransport.swift#L369

- The `peerLeft`, not yet, because I have not implemented on iOS the reconnection logic yet, which we currently have on web. This is currently something pending in my TODO list. 🙂